### PR TITLE
Add a ton of new emotes + some small species emote refactors

### PR DESCRIPTION
**Pull request by itsmeow**
## About The Pull Request

Re-PR of #7645

Self-credit for the annoyed sprite.

Credit to linked Skyrat PR for sigh, glassses, sweatdrop, and the idea.

Partial port of https://github.com/Skyrat-SS13/Skyrat-tg/pull/6259
Ports https://github.com/shiptest-ss13/Shiptest/pull/972
Moth laugh (as far as I can tell?) is from https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10768

Improved the animated icons sprites, fixed the animation looping, and refactored almost all of the emotes to properly check species requirements.

Species now have a get_laugh_sound and get_scream_sound proc that is used for the emotes instead of doing typechecks.

Same for cough, gasp, sigh, sneeze, and sniff.

Added a lot of organ and species checks for existing emotes.

### Original PR Description

    Emotes added/improved with sounds:
    *cough - 3 female and male variants
    *dwoop - Silicon and IPC only
    *gasp - 6 female and male variants
    *msqueak - Moth and mothroach only
    Unique Moth laugh soound
    *chitter - Moth and mothroach only
    *sigh - One female and male variants
    *snap2 - Snaps twice
    *snap3 - Snaps thrice
    *sneeze - female and male variant
    *sniff - female and male variant

    Emotes added without sounds:
    *clear - "Clears their throat"
    *etwitch - "Twitches their ears"
    *eyeroll - "Rolls their eyes"
    *huff - "lets out a huff!"

    ANIMATED emotes added with sounds:
    *sweatdrop
    *glasses (must have any glasses on the eye slot)
    *annoyed
    *sigh

## Why It's Good For The Game

More emotes = more expression and RP (I hope), also fixes a bunch of emotes being usable when they really shouldn't.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

(removed exclaim and question)

https://user-images.githubusercontent.com/10366817/196009269-f65e14a7-37c8-4ae0-96ca-47dd2a8a3130.mp4

test

test

test

test

</details>

## Changelog
:cl:
add: Added animated emotes: *sweatdrop, *glasses, *annoyed, *sigh
add: Added audible emotes: *cough, *dwoop (IPC), *gasp, *msqueak (moth), *chitter (moth), *sigh, *snap2, *snap3, *sneeze, *sniff
add: Added emotes: *clear, *etwitch (felinid), *eyeroll, *huff
tweak: Abductors, androids, apids, flypeople, golems, IPCs, moths, skeletons, and non-/mob/living/carbon/humans, and anyone with the NOBREATH trait will not play a sound for *cough, *gasp, *sigh, *sneeze, and *sniff
add: Added a unique moth laugh sound.
add: Added a unique moth deathgasp sound.
tweak: Mothroaches now use insect-type emote messages and can use *click, *chitter, and *msqueak.
code: Clap emote now properly checks if you have arms.
code: Improved how *laugh and *scream determine per-species sounds.
fix: Removed a duplicate *chime emoji.
fix: Fixed some emotes having an extra period at the end of their messages.
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
Needs some changes before merge

**Comment by itsmeow** [comment]
All fixed.

File: code/modules/mob/living/carbon/emote.dm [hunk 1]
```diff
@@ -77,10 +95,43 @@
 				'sound/creatures/monkey/monkey_screech_6.ogg',
 				'sound/creatures/monkey/monkey_screech_7.ogg')
 
+/datum/emote/living/carbon/snap
```
Comment (dwasint) [inline code/modules/mob/living/carbon/emote.dm:98]: shouldn't snaps check if you have atleast one usable hand so armless people can't snap?

Comment (itsmeow) [reply code/modules/mob/living/carbon/emote.dm:98]: The restraint_check does that I think, but I suppose checking for arm organs could make sense

**Comment by PowerfulBacon** [comment]
I'm not really a fan of the blue emote overhead text either to be honest, but I guess thats an opinion. Show a before and after of that change in the screenshots section.

**Comment by PowerfulBacon** [comment]
image
This looks bad and just seems way too big. I can't really think of a specific reason to dislike them apart from it being a big OOC overlay that looks out of place, but they still just seem off. Especially the giant question mark.

File: code/modules/mob/living/emote.dm [hunk 2]
```diff
@@ -4,6 +4,20 @@
...
+	message = "bleps their tongue out"
+	message_AI = "shows an image of a random blepping animal"
+	vary = TRUE
+	mob_type_blacklist_typecache = list(/mob/living/carbon/alien,/mob/living/silicon/robot)
+
+/datum/emote/living/blep/can_run_emote(mob/user, status_check = TRUE, intentional)
+	if(!..())
+		return FALSE
+	var/obj/item/organ/tongue/T = user.getorganslot(ORGAN_SLOT_TONGUE)
+	return isAI(user) || istype(T)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/emote.dm]:
```
X bleps their tongue out.
```
What is this even meant to mean? Emotes being added should actually make sense and be useful, we already have loads that should just be a *me emote.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 3]
```diff
@@ -14,6 +14,18 @@
...
+	key = "etwitch"
+	key_third_person = "twitches their ears"
+	message = "twitches their ears"
+	vary = TRUE
+
+/datum/emote/living/carbon/human/etwitch/can_run_emote(mob/user, status_check = TRUE, intentional)
+	if(!..())
+		return FALSE
+	var/mob/living/carbon/human/H = user
+	return ("ears" in H.dna?.species?.mutant_bodyparts)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/emote.dm:27]: How do you twitch your ears, and when would this emote ever be useful?

File: code/modules/mob/living/carbon/emote.dm [hunk 4]
```diff
@@ -110,3 +162,65 @@
...
+	sound_volume = 25
+	vary = TRUE
+	overlay_x_offset = 10
+	overlay_y_offset = 10
+	emote_length = 3 SECONDS
+	sound = 'code/datums/emote_sounds/emotes/question.ogg'
+
+/datum/emote/living/carbon/annoyed
+	key = "annoyed"
+	key_third_person = "is annoyed"
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/emote.dm]: I don't think this is how key_third_person works, is it not an alias for key and thus should be a single word?

File: code/modules/mob/living/carbon/emote.dm [hunk 5]
```diff
@@ -77,10 +95,44 @@
...
+	message = "snaps their fingers twice"
+	message_param = "snaps their fingers at %t twice"
+	sound = 'code/datums/emote_sounds/emotes/snap2.ogg'
+
+/datum/emote/living/carbon/snap/three
+	key = "snap3"
+	key_third_person = "snaps thrice"
+	message = "snaps their fingers thrice"
+	message_param = "snaps their fingers at %t thrice"
+	sound = 'code/datums/emote_sounds/emotes/snap3.ogg'
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/emote.dm]: These don't really need to exist, I can't see them being used a lot

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
See above, some of the sprites are still pretty out of place I think, and the PR body needs more screenshots and preferably videos of anything animated

**Comment by PowerfulBacon** [comment]
If anything, the emotes that display should be some kind of overlay to the character, for example a sad/defeated emote could have a blue gradiant from the top of the characters head rather than a giant raincloud appear above their head or something. Gives it more of an immersive vibe which is pretty good if you want an atmosphere that supports RP rather than little sprinkles that give the illusion of RP getting better.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 3 above)
Comment (Fronsis) [reply code/modules/mob/living/carbon/human/emote.dm:27]: Probably intended for the races that have a bigger and more controllable ears like catpeople, lizards, etc, while i know people that can move their ears  IRL.. it's a little strange yeah

File: code/modules/mob/living/carbon/emote.dm (hunk 5 above)
Comment (Fronsis) [reply code/modules/mob/living/carbon/emote.dm]: I've personally played  and observed on the servers that have snap2 and snap3 and.. it gets used, it's not used as in ''haha fun snap3.ogg spam'' but more like... i've seen it used on situations where snap2 is used like ''go on GET TO MEDBAY NOW CHOP CHOP snap2'' or snap3 when someone makes snarky remarks and so on, for the regular spessman that doesn't try to roleplay, they won't give it use but having it available as another option for expressing something or to add it to follow a sentence feels nice

File: code/modules/mob/living/carbon/human/emote.dm (hunk 3 above)
Comment (itsmeow) [reply code/modules/mob/living/carbon/human/emote.dm:27]: It only works on mutant ears, so just felinids

File: code/modules/mob/living/emote.dm (hunk 2 above)
Comment (itsmeow) [reply code/modules/mob/living/emote.dm]: Removed

File: code/modules/mob/living/carbon/human/emote.dm (hunk 3 above)
Comment (itsmeow) [reply code/modules/mob/living/carbon/human/emote.dm:27]: Cats IRL twitch their ears to hear sounds when on alert or curious,

File: code/modules/mob/living/carbon/emote.dm (hunk 5 above)
Comment (itsmeow) [reply code/modules/mob/living/carbon/emote.dm]: Yeah these are useful imo

**Comment by itsmeow** [comment]
Uploaded a video

**Comment by PowerfulBacon** [comment]
Hm not really a fan of the icon based emotes, since they are almost like meta-elements like UI overlays displayed in-world. Interested in what other people think though

**Comment by itsmeow** [comment]
I don't really like the color change tbh I am going to revert it

File: code/modules/mob/living/emote.dm [hunk 6]
```diff
@@ -427,10 +389,8 @@
 		if(custom_emote && !check_invalid(user, custom_emote))
 			var/type = input("Is this a visible or hearable emote?") as null|anything in list("Visible", "Hearable")
 			switch(type)
-				if("Visible")
-					emote_type = EMOTE_VISIBLE
```
Comment (itsmeow) [inline code/modules/mob/living/emote.dm:431]: bug

**Comment by Bokkiewokkie** [comment]
Shouldn’t the moth emotes be available to other bug species like flypeople and apids as well?

**Comment by itsmeow** [comment]
As someone who doesn't watch any anime I don't really connect any of them to anime. More just a replacement for our inability to create facial expressions.

**Comment by EvilDragonfiend** [comment]
> More just a replacement for our inability to create facial expressions.

Neither I don't connect any of them to facial expressions. These are symbols, not your real facial expressions.

**Comment by DrDuckedGoose** [comment]
That gasp...

**Comment by Megaddd** [comment]
I'm confused, is it just the *emotes being shown in runechat now? Or are the weird anime angry sprites and exclamations, questionmarks making it in as well in the screenshots below the video?

Showing *me and similar emotes in runechat is a great improvement.
Auxiliary sprites next to the character when emoting not so much.

**Comment by itsmeow** [comment]
> Shouldn’t the moth emotes be available to other bug species like flypeople and apids as well?

ehhh they're all directly based off the moth scream which only moths have

**Comment by itsmeow** [comment]
PR is all fixed up, I removed question and exclaim since I don't want to make overlay sprites at the moment. I also changed the emote text color back to gray.

**Comment by Tsar-Salat** [comment]
> PR is all fixed up, I removed question and exclaim since I don't want to make overlay sprites at the moment. I also changed the emote text color back to gray.

cant you just reuse the exclamation from when someone points a gun, gives an item, or reveals themself from a box (le funny metal gear reference

**Review by PowerfulBacon — DISMISSED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**